torch/csrc/jit/passes/peephole.cpp

```diff
@@ -244,14 +244,15 @@ void PeepholeOptimizeImpl(Block* block, bool addmm_fusion_enabled) {
 
 void PeepholeOptimize(Block* block, bool addmm_fusion_enabled) {
   PeepholeOptimizeImpl(block, addmm_fusion_enabled);
-  // Eliminate dead code created by any peephole passes we've just done
-  EliminateDeadCode(block);
 }
 
 void PeepholeOptimize(
     const std::shared_ptr<Graph>& graph,
     bool addmm_fusion_enabled) {
   PeepholeOptimize(graph->block(), addmm_fusion_enabled);
+  // Eliminate dead code created by any peephole passes we've just done,
+  // unsafe to only run on block
```

Comment:
You need the complete graph in order to track aliasing information, so that you do not DCE a mutating op of a value that is then used later in the graph. There is some logic in DCE to act conservatively in these situations but I'm not sure how well tested it is.

In the previous version, when I had prim::sort as an unschematized op that was handled in shape_analysis it was DCEing it. This may be the only case it can't handle but in general still bad practice to DCE just the block. 

@suo

Reply:
DCE will work correctly with just a block but mutable ops will never get removed, even if there are no live uses of any alias.

Reply:
it was removing the unschematized mutating op